### PR TITLE
Bump up the version for ROCm-6.2.0

### PR DESCRIPTION
**Pull request by afzpatel**

### CODE REVIEW

**Comment by afzpatel** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/composable-kernel/package.py [hunk 1]
```diff
@@ -88,11 +90,11 @@ def cmake_args(self):
         ]
         if "auto" not in self.spec.variants["amdgpu_target"]:
             args.append(self.define_from_variant("GPU_TARGETS", "amdgpu_target"))
+        else:
+            args.append(self.define("INSTANCES_ONLY", "ON"))
```
Comment (renjithravindrankannath) [inline var/spack/repos/builtin/packages/composable-kernel/package.py:94]: args.append(self.define("INSTANCES_ONLY", True))

**Comment by eugeneswalker** [comment]
As soon as this is merged, we will update E4S's External ROCm CI stack to use ROCm 6.2.0! Thank you!

**Review by srekolam — COMMENTED** [review]
LGTM. Thanks,

**Review by renjithravindrankannath — APPROVED** [review]

**Comment by afzpatel** [comment]
Hi @tldahlgren, this pr is ready to be reviewed.

**Review by eugeneswalker — DISMISSED** [review]

**Comment by eugeneswalker** [comment]
Does anyone else want to do final review before we merge? @srekolam I see some changes pushed after your last approval-- does it still look good? Thanks!

**Comment by tldahlgren** [comment]
> Does anyone else want to do final review before we merge? @srekolam I see some changes pushed after your last approval-- does it still look good? Thanks!

I plan to after my meetings are over.

File: var/spack/repos/builtin/packages/miopen-hip/package.py [hunk 2]
```diff
@@ -84,12 +85,31 @@ class MiopenHip(CMakePackage):
...
+        "5.7.0",
+        "5.7.1",
+        "6.0.0",
+        "6.0.2",
+        "6.1.0",
+        "6.1.1",
+        "6.1.2",
+    ]:
+        depends_on(f"rocm-clang-ocl@{ver}", when=f"@{ver}")
+
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/miopen-hip/package.py:112]: Does this change mean this package no longer depends on `rocm-clang-ocl` for its versions `@6.2.0:`?

File: var/spack/repos/builtin/packages/miopen-hip/package.py [hunk 3]
```diff
@@ -174,14 +197,21 @@ def cmake_args(self):
             args.append(self.define("MIOPEN_USE_MLIR", "OFF"))
         if self.spec.satisfies("@5.7.0:"):
             args.append(self.define("MIOPEN_ENABLE_AI_IMMED_MODE_FALLBACK", "OFF"))
-        args.append(
-            "-DNLOHMANN_JSON_INCLUDE={0}".format(self.spec["nlohmann-json"].prefix.include)
-        )
-        if self.spec.satisfies("@6.0.0:"):
+        if self.spec.satisfies("@6.0.0:6.1"):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/miopen-hip/package.py]: FWIW You could simplify this:

```suggestion
        if self.spec.satisfies("@6:6.1"):
```

**Review by tldahlgren — DISMISSED** [review]
Confirmed the homepage of the new package and all new version (directive) sha256s.  Have a quick question about a change and a very minor suggestion.

File: var/spack/repos/builtin/packages/miopen-hip/package.py (hunk 2 above)
Comment (afzpatel) [reply var/spack/repos/builtin/packages/miopen-hip/package.py:112]: Yes, this package will no longer depend on rocm-clang-ocl from 6.2 onwards. Rocm-clang-ocl is deprecated now, 6.1.2 was the final release.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/miopen-hip/package.py:112]: Thanks for the confirmation.

**Review by tldahlgren — DISMISSED** [review]

**Review by eugeneswalker — DISMISSED** [review]

File: var/spack/repos/builtin/packages/rpp/package.py [hunk 4]
```diff
@@ -152,7 +153,7 @@ def patch(self):
 
     with when("+hip"):
         with when("@5.7:"):
-            for ver in ["5.7.0", "5.7.1", "6.0.0", "6.0.2", "6.1.0", "6.1.1", "6.1.2"]:
+            for ver in ["5.7.0", "5.7.1", "6.0.0", "6.0.2", "6.1.0", "6.1.1", "6.1.2", "6.2.0"]:
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/rpp/package.py:156]: can you please have the new version of boost . depends_on("boost@1.72.0:1.85.0") in the rpp package.

**Review by srekolam — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/rpp/package.py (hunk 4 above)
Comment (afzpatel) [reply var/spack/repos/builtin/packages/rpp/package.py:156]: done

**Review by eugeneswalker — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]

**Comment by balay** [comment]
I'm seeing the following behavior with this change.
```
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack.a$ ./bin/spack spec ginkgo+rocm amdgpu_target=gfx90a ^hip@5.7.1 |egrep '(hip|hipfft)@'
 -       ^hip@5.7.1
 -       ^hip@5.7.1%gcc@11.4.0~asan~cuda~ipo+rocm build_system=cmake build_type=Release generator=make patches=5bb9b0e,7668b2a,aee7249,b589a02,c2ee21c arch=linux-ubuntu22.04-x86_64
 -       ^hipfft@master%gcc@11.4.0~asan~cuda~ipo+rocm amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64
```
notice: `hipfft@master`. However with the prior commit, I see: `^hipfft@5.7.1` gets used:
```
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack.a$ ./bin/spack spec ginkgo+rocm amdgpu_target=gfx90a ^hip@5.7.1 |egrep '(hip|hipfft)@'
 -       ^hip@5.7.1
 -       ^hip@5.7.1%gcc@11.4.0~asan~cuda~ipo+rocm build_system=cmake build_type=Release generator=make patches=5bb9b0e,7668b2a,aee7249,b589a02,c2ee21c arch=linux-ubuntu22.04-x86_64
 -       ^hipfft@5.7.1%gcc@11.4.0~asan~cuda~ipo+rocm amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64
```

**Comment by balay** [comment]
> notice: `hipfft@master`

potential fix (workaround?) at https://github.com/spack/spack/pull/46376

**Pull request merged**